tests/integration/flows/test_bigquery_to_adls.py

```diff
@@ -1,23 +1,70 @@
 from viadot.flows import BigQueryToADLS
 from prefect.tasks.secrets import PrefectSecret
+import pendulum
+import os
+
+ADLS_PATH = "raw/tests"
+FILE_NAME = str(pendulum.now("utc")) + ".parquet"
+BIGQ_CREDENTIAL_KEY = "BIGQUERY_TESTS"
+ADLS_CREDENTIAL_SECRET = PrefectSecret(
+    "AZURE_DEFAULT_ADLS_SERVICE_PRINCIPAL_SECRET"
+).run()
 
 
 def test_bigquery_to_adls():
-    credentials_secret = PrefectSecret(
-        "AZURE_DEFAULT_ADLS_SERVICE_PRINCIPAL_SECRET"
-    ).run()
+    flow_bigquery = BigQueryToADLS(
+        name="Test BigQuery to ADLS extract",
+        dataset_name="official_empty",
+        table_name="space",
+        adls_file_name=FILE_NAME,
+        credentials_key=BIGQ_CREDENTIAL_KEY,
+        adls_dir_path=ADLS_PATH,
+        adls_sp_credentials_secret=ADLS_CREDENTIAL_SECRET,
+    )
+
+    result = flow_bigquery.run()
+    assert result.is_successful()
+
+    task_results = result.result.values()
+    assert all([task_result.is_successful() for task_result in task_results])
+    os.remove("test_bigquery_to_adls_extract.parquet")
+    os.remove("test_bigquery_to_adls_extract.json")
+
 
+def test_bigquery_to_adls_overwrite_true():
     flow_bigquery = BigQueryToADLS(
-        name="BigQuery to ADLS",
+        name="Test BigQuery to ADLS overwrite true",
         dataset_name="official_empty",
         table_name="space",
-        credentials_key="BIGQUERY_TESTS",
-        adls_dir_path="raw/tests",
-        adls_sp_credentials_secret=credentials_secret,
+        credentials_key=BIGQ_CREDENTIAL_KEY,
+        adls_file_name=FILE_NAME,
+        overwrite_adls=True,
+        adls_dir_path=ADLS_PATH,
+        adls_sp_credentials_secret=ADLS_CREDENTIAL_SECRET,
     )
 
     result = flow_bigquery.run()
     assert result.is_successful()
 
     task_results = result.result.values()
     assert all([task_result.is_successful() for task_result in task_results])
+    os.remove("test_bigquery_to_adls_overwrite_true.parquet")
+    os.remove("test_bigquery_to_adls_overwrite_true.json")
+
+
+def test_bigquery_to_adls_false():
+    flow_bigquery = BigQueryToADLS(
+        name="Test BigQuery to ADLS overwrite false",
+        dataset_name="official_empty",
+        table_name="space",
+        adls_file_name=FILE_NAME,
+        overwrite_adls=False,
+        credentials_key=BIGQ_CREDENTIAL_KEY,
+        adls_dir_path=ADLS_PATH,
+        adls_sp_credentials_secret=ADLS_CREDENTIAL_SECRET,
+    )
+
+    result = flow_bigquery.run()
+    assert result.is_failed()
```

Comment:
You can also check why flow failed i.e. if "FileExistsError: File already exists!" shows up in logs